tests/test_registrable.py

```diff
@@ -160,3 +169,26 @@ class Test(Registrable):
     # Function returns class name if no entry in registry was found
     assert A.name_for_object_class(c) == "C"
     assert A.name_for_object_class(test) == "Test"
+
+
+def test_auto_classes():
+    class Base(Registrable):
+        pass
+
+    T = TypeVar("T", bound=RegistrableProtocol)
+
+    class Auto(Registrable[T]):
+        pass
+
+    class AutoBase(Auto[Base]):
+        BASE_CLASS = Base
```

Comment:
Why `AutoBase` (in addition to "Auto") at all? Can you simplify the whole test setup?